pkg/core/writer/splunk/splunk.go

```diff
@@ -0,0 +1,221 @@
+package splunk
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/signalfx/golib/v3/datapoint"
+	"github.com/signalfx/golib/v3/event"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+)
+
+// Handler posts data to Splunk HTTP Event Collector.
+type Handler struct {
+	HTTPClient    *http.Client
+	URL           string
+	Hostname      string
+	Token         string
+	Source        string
+	SourceType    string
+	Index         string
+	SkipTLSVerify bool
+	Errors        chan error
+	once          sync.Once
+	events        chan splunkMessage
+	sendQueue     chan []splunkMessage
+}
+
+func (h *Handler) init() {
+	if h.HTTPClient == nil {
+		tr := &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: h.SkipTLSVerify}}
```

Comment:
Use https://github.com/signalfx/signalfx-agent/blob/master/pkg/core/common/httpclient/http.go which has a common config struct and building for HTTP clients.

Using transport without setting things like MaxIdleConns can cause issues so it's best to clone the default transport which the builder does.

Reply:
Thanks for the review! I have added a commit following your recommendation.